# chi_square_test

## Association between Device Type and Customer Satisfaction

### Objective:
To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.

In [9]:
import numpy as np
import pandas as pd
from scipy.stats import chi2_contingency, chi2

In [10]:
#  Create Contingency Table
data = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])

## State Hypotheses
#### H0: No association between device type and customer satisfaction
#### H1: Significant association exists

## Perform Chi-Square Test

In [11]:
chi2_stat, p_val, dof, expected = chi2_contingency(data)

In [12]:
print("Chi-Square Statistic:", chi2_stat)
print("p-value:", p_val)
print("Degrees of Freedom:", dof)
print("Expected Frequencies:\n", expected)

Chi-Square Statistic: 5.638227513227513
p-value: 0.22784371130697179
Degrees of Freedom: 4
Expected Frequencies:
 [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


## Determine the Critical Value

In [13]:
#  Determine Critical Value at α = 0.05
alpha = 0.05
critical_value = chi2.ppf(1 - alpha, dof)

print("Critical Value:", critical_value)

Critical Value: 9.487729036781154


## Make a Decision

In [14]:
#  Make Decision
if chi2_stat > critical_value:
    print("Reject Null Hypothesis: Significant association exists.")
else:
    print("Fail to Reject Null Hypothesis: No significant association.")


Fail to Reject Null Hypothesis: No significant association.


## Explanation of Steps:
### Hypotheses:

H₀: No association between device type and satisfaction.

H₁: Significant association exists.

Chi-Square Test checks if observed counts differ significantly from expected counts under independence.

Degrees of Freedom (df): (rows−1)×(columns−1)=(5−1)×(2−1)=4

### Critical Value:
From Chi-square table at df=4, α=0.05.

### Decision Rule:

If test statistic > critical value → Reject H₀

Else → Fail to reject H₀

## Conclusion:
Since Chi-Square Statistic (5.64) < Critical Value (9.48),
we fail to reject the null hypothesis.
=> No significant association exists between the type of device and customer satisfaction at 5% significance level.